### ISSUE TITLE
Improve UX

### ISSUE THREAD

**Issue by pax-k** [aws/copilot-cli#4892]
Hey! Killing the command with CTRL+C while it provisions resources gives me a hard time when I try to work with copilot. Trying to redeploy / do stuff around yields errors. I have to manually delete apps / service / envs, or to kill Cloudformation stacks manually. Would it be an idea for the CLI to listen for CTLR+C and trigger a rollback?

Also it's not yet clear for me how to handle multiple apps from a monorepo. Should the copilot directory be at the root of the monorepo or at the root of each app?

[Also this https://github.com/aws/copilot-cli/issues/3676](https://github.com/aws/copilot-cli/issues/3676#issuecomment-1553372063)](https://github.com/aws/copilot-cli/issues/3676#issuecomment-1553372063)

Also please add `remove` alongside `delete`. Remove should just soft-delete / remove aws resources, without deleting any local files - leave that to `delete`.

Also the official documentation uses the words "app" / "service" interchangeably like they mean the same thing: https://aws.github.io/copilot-cli/docs/commands/svc-init/

> After running this command, the CLI creates sub-directory with your app name in your local copilot directory where you'll find a manifest file.

No, "the CLI creates sub-directory with your **service** name."

The `copilot svc init` command should also ask for build context, besides Dockerfile path. 

And even if I previously created a dev environment, the `copilot svc init` command after it finishes still recommends test environment:
```- Run copilot svc deploy --name gateway --env test` to deploy your service to a test environment.```

So this is the full chain of commands I have to run if I use a custom domain:
```
copilot app init some-app --domain domain.xyz
copilot init
copilot env init
copilot env deploy --name dev
copilot svc init --name gateway --svc-type "Load Balanced Web Service"
copilot svc deploy --name gateway --env dev
```
Notice the first 2 commands, the order feels bad, but this is what the docs say: https://aws.github.io/copilot-cli/docs/commands/app-init/

If any of these steps fail or are killed while provisioning, I have a hard time rolling back.
I would love to be able to add multiple apps to the same copilot/ directory and compress everything into 2 commands:

- `copilot init`: should ask for app name, domain if necessary, then for environment name, then for service name
- `copilot deploy`: should ask what app context to use, what service, what environment, done

If it happens that I trigger CTRL+C, the script should trigger a rollback.
If some CloudFormation tasks fails, I want to see reason and the logs.
If I don't get the commands order right from whatever reason, stuff gets provisioned, then the command eventually fails. Because there were insufficient checks implemented in the first place

So far I wasted my entire day wrapping my head around all these copilot quirks, sorry. Not to mention that CloudFormation provisioning takes ages.

Also I'm trying to deploy a service `copilot svc deploy --name gateway --env dev` and I have no clue why it fails 10 times because I see 0 obvious logs. Eventually I found the real reason hidden somewhere in aws:
```
ResourceInitializationError: unable to pull secrets or registry auth: execution resource retrieval failed: unable to retrieve secrets from ssm: service call has been retried 1 time(s): AccessDeniedException: User: arn:aws:sts::x:assumed-role/x-ExecutionRole-x/x is not authorized to perform: ssm:GetParameters on resource: arn:aws:ssm:eu-north-1:x:parameter/API_KEY because no identity-based policy allows the ssm:GetParameters action status code: 400, request id: x
```
I referenced a secret without going through `copilot secret init`. Now that I did that, I repeated the service deployment but it's stuck in the previous deployment.

Again, implementing checks before actual deployments would save a lot of time.

Not trying to be an as***e, but there is zero UX for the CLI

**Comment by dannyrandall** [comment]
Hey @pax-k! Thanks for all the feedback.

> Killing the command with CTRL+C

Does #2544 look like what you're thinking about? If so, please thumbs-up there and/or add any additional context!!

> add `remove` alongside `delete`

I'm not quite sure I follow the difference - right now, running `copilot svc delete` _does not_ delete your local manifest files.

> official documentation uses the words "app" / "service"

Good catch 😊 @huanjani fixed that in #4894!

> The `copilot svc init` command should also ask for build context

There's a tradeoff here - not every manifest needs `context` to be set, it just depends on how your repo and Dockerfiles are structured. Ideally, my opinion is that `copilot svc init` should minimize the number of questions asked, especially questions that aren't relevant to every configuration.

> it finishes still recommends test environment

This is a great one - Copilot definitely should find an existing environment, if one exists, and substitute that for `test`. 👍

> Notice the first 2 commands, the order feels bad

I'm not sure where you're seeing that the correct order is `copilot app init` -> `copilot init`. Your order was correct other than `copilot init` - it's not necessary. I think we could remove the section saying that `copilot app init` and `copilot init` are interchangeable, though. `copilot init` is a single command to create an app, environment, and service in a single command.

> compress everything into 2 commands

I like the `copilot deploy` step to deploy all of the services in the repo!

> I have no clue why it fails 10 times because I see 0 obvious logs

We should find a nice way to surface application/ECS logs within our progress tracker! It _is_ difficult to troubleshoot errors that involve the circuit breaker. I think surfacing those logs along side something like https://github.com/aws/copilot-cli/issues/2041 might be nice here.

**Comment by KollaAdithya** [comment]
Hello @pax-k  !
Ctrl-C feature is released in [v1.30](https://github.com/aws/copilot-cli/releases/tag/v1.30.0) 🎉 . You can now cancel your stack update to rollback to previous configuration.

**Comment by Lou1415926** [comment]
> We should find a nice way to surface application/ECS logs within our progress tracker! It is difficult to troubleshoot errors that involve the circuit breaker. I think surfacing those logs along side something like https://github.com/aws/copilot-cli/issues/2041 might be nice here.


For anyone that comes across this - this enhancement ⬆️  is already released in AWS Copilot v1.31.0! For more, see the [release note](https://github.com/aws/copilot-cli/releases/tag/v1.31.0) and the [blog post](https://aws.github.io/copilot-cli/blogs/release-v131/) 🎉🚀 🚀

**Comment by dannyrandall** [comment]
We also released an enhanced `copilot deploy` command in [v1.30](https://aws.github.io/copilot-cli/blogs/release-v130/#copilot-deploy-enhancements) and [v1.31](https://aws.github.io/copilot-cli/blogs/release-v131/#copilot-deploy-enhancements). With that, I think the relevant feedback in this issue has been resolved so I'll go ahead and close it! Feel free to reopen/open a new issue if you have other feedback.
